ament_cmake_version/cmake/version.h.in

```diff
@@ -0,0 +1,34 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ${PROJECT_NAME_UPPER}__VERSION_H_
+#define ${PROJECT_NAME_UPPER}__VERSION_H_
+
+/// \def ${PROJECT_NAME_UPPER}_VERSION_MAJOR
+/// Defines ${PROJECT_NAME_UPPER} major version number
+#define ${PROJECT_NAME_UPPER}_VERSION_MAJOR (${VERSION_MAJOR})
+
+/// \def ${PROJECT_NAME_UPPER}_VERSION_MINOR
+/// Defines ${PROJECT_NAME_UPPER} minor version number
+#define ${PROJECT_NAME_UPPER}_VERSION_MINOR (${VERSION_MINOR})
+
+/// \def ${PROJECT_NAME_UPPER}_VERSION_PATCH
+/// Defines ${PROJECT_NAME_UPPER} version patch number
+#define ${PROJECT_NAME_UPPER}_VERSION_PATCH (${VERSION_PATCH})
+
+/// \def ${PROJECT_NAME_UPPER}_VERSION_STR
+/// Defines ${PROJECT_NAME_UPPER} version string
+#define ${PROJECT_NAME_UPPER}_VERSION_STR "${VERSION_STR}"
```

Comment:
Also add add `${PROJECT_NAME_UPPER}_VERSION_GTE(major, minor, patch)` macro as defined in https://github.com/ros2/rmw_cyclonedds/pull/51/files#diff-d03c99cafce317e1a93de46e787e0235R64?

```cpp
/// \def ${PROJECT_NAME_UPPER}_VERSION_GTE
/// Defines a macro to check whether the version of ${PROJECT_NAME_UPPER} is greater than or equal to
/// the given version triple.
#define ${PROJECT_NAME_UPPER}_VERSION_GTE(major, minor, patch) ( \
     (major < ${PROJECT_NAME_UPPER}_VERSION_MAJOR) ? true \
     : (major > ${PROJECT_NAME_UPPER}_VERSION_MAJOR) ? false \
     : (minor < ${PROJECT_NAME_UPPER}_VERSION_MINOR) ? true \
     : (minor > ${PROJECT_NAME_UPPER}_VERSION_MINOR) ? false \
     : (patch < ${PROJECT_NAME_UPPER}_VERSION_PATCH) ? true \
     : (patch > ${PROJECT_NAME_UPPER}_VERSION_PATCH) ? false \
     : true)
```

I could use such a macro already for `rosidl_generator_cpp`, to compile my ROS 2 code conditionally depending on the version of [rosidl](https://github.com/ros2/rosidl). `rosidl_generator_cpp::MessageInitialization` and other types have been moved to namespace `rosidl_runtime_cpp` in [version 0.9.0](https://github.com/ros2/rosidl/blob/master/rosidl_generator_cpp/CHANGELOG.rst#090-2020-04-24), and I do not see another way to use a non-default message initialization flag such that it works in dashing, eloquent and foxy.

For this specific case it would be convenient to have a
```cpp
using MessageInitialization = rosidl_runtime_cpp::MessageInitialization;
```
declaration within the generated message structs, but that is an issue for https://github.com/ros2/rosidl.

Reply:
@meyerj, done, see the last commit https://github.com/ament/ament_cmake/pull/198/commits/1b110bdea618f99b16ef61063d7e23aebdcc59ae